plugins/inputs/mock/mock.go

```diff
@@ -0,0 +1,160 @@
+package mock
+
+import (
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Mock struct {
+	counter int64
+
+	MetricName string            `json:"metric_name"`
+	Tags       map[string]string `json:"tags"`
+
+	Random   []*Random   `json:"random_float"`
+	Step     []*Step     `json:"step"`
+	Stock    []*Stock    `json:"stock"`
+	SineWave []*SineWave `json:"sine_wave"`
+}
+
+type Random struct {
+	Name string  `json:"name"`
+	Min  float64 `json:"min"`
+	Max  float64 `json:"max"`
+}
+
+type SineWave struct {
+	Name      string  `json:"name"`
+	Amplitude float64 `json:"amplitude"`
+}
+
+type Step struct {
+	latest float64
+
+	Name  string  `json:"name"`
+	Start float64 `json:"min"`
+	Step  float64 `json:"max"`
+}
+
+type Stock struct {
+	latest float64
+
+	Name       string  `json:"name"`
+	Price      float64 `json:"price"`
+	Volatility float64 `json:"volatility"`
+}
+
+const SampleConfig = `
```

Comment:
Please do not export this constant
```suggestion
const sampleConfig = `
```